### PR TITLE
Elie.abichar/rumm 3337 improve documentation

### PR DESCRIPTION
**Pull request by eliac123**
### What does this PR do?

Improve documentation and bump react native version to 71.10

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [ ] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

**Comment by rtrieu** [comment]
Created [DOCS-5672](https://datadoghq.atlassian.net/browse/DOCS-5672) for docs review.

File: rum-expo-react-navigation/README.md
Comment (0xnm) [inline rum-expo-react-navigation/README.md:38]:
```suggestion
We should note that the React Native SDK doesn't really run on Expo Go, so to have an honest build, you need to build the project using `expo-dev-client`. You can do that by running:
```

File: rum-expo-react-navigation/README.md
Comment (0xnm) [inline rum-expo-react-navigation/README.md:50]:
```suggestion
### On EAS
```

File: rum-expo-react-navigation/README.md
Comment (rtrieu) [inline rum-expo-react-navigation/README.md:38]:
```suggestion
The React Native SDK runs separately from Expo Go, which means the project should be built using `expo-dev-client`. Build the project locally by running the following:
```

File: rum-expo-react-navigation/README.md
Comment (rtrieu) [inline rum-expo-react-navigation/README.md:54]:
```suggestion
Learn how to sign the app in the [EAS documentation][1].
```

File: rum-react-navigation-async/README.md
Comment (rtrieu) [inline rum-react-navigation-async/README.md:54]:
```suggestion
**Note**: The [Datadog API KEY][2] must be valid.
```

File: rum-react-navigation-async/README.md
Comment (rtrieu) [inline rum-react-navigation-async/README.md:46]:
```suggestion
Since we have added the [`datadog-react-native-wizard`][1], you need to add the `datadog-ci.json` file in the project root that includes the following:
```

File: rum-react-navigation-codepush/README.md
Comment (rtrieu) [inline rum-react-navigation-codepush/README.md:37]:
```suggestion
Since we have added the [`datadog-react-native-wizard`][1], you need to add the `datadog-ci.json` file in the project root that includes the following:
```

File: rum-react-navigation-codepush/README.md
Comment (rtrieu) [inline rum-react-navigation-codepush/README.md:45]:
```suggestion
**Note**: The [Datadog API KEY][2] must be valid.
```

File: rum-react-navigation/README.md
Comment (rtrieu) [inline rum-react-navigation/README.md:36]:
```suggestion
Since we have added the [`datadog-react-native-wizard`][1], you need to add the `datadog-ci.json` file in the project root that includes the following:
```

File: rum-react-navigation/README.md
Comment (rtrieu) [inline rum-react-navigation/README.md:44]:
```suggestion
**Note**: The [Datadog API KEY][2] must be valid.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Some suggestions for consistency and conciseness, but otherwise looks good!

**Comment by rtrieu** [comment]
Hi @eliac123 , waiting for you review the suggestions before approving this.

**Comment by rtrieu** [comment]
Hi @0xnm, I went ahead and committed my own suggestions because they were for style/consistency and not the content. If the rest of it looks good to you, we can merge this!

**Review by louiszawadzki — APPROVED** [review]

File: rum-expo-react-navigation/README.md
Comment (rtrieu) [inline rum-expo-react-navigation/README.md:38]:
```suggestion
The React Native SDK runs separately from Expo Go, which means the project must be built using `expo-dev-client`. Build the project locally by running the following:
```

**Review by rtrieu — APPROVED** [review]
Approving with one suggestion!

**Pull request merged**
